infra/main.bicep

```diff
@@ -0,0 +1,421 @@
+
+@description('The pricing tier for the App Service plan')
+@allowed([
+  'F1'
+  'D1'
+  'B1'
+  'B2'
+  'B3'
+  'S1'
+  'S2'
+  'S3'
+  'P1'
+  'P2'
+  'P3'
+  'P4'
+])
+param hostingPlanSku string = 'B3'
+
+@description('Name of Web App')
+param websiteName string = '${resourcePrefix}-website'
+
+@description('Name of Application Insights')
+param applicationInsightsName string = '${resourcePrefix}-appinsights'
+
+@description('Use semantic search')
+param azureSearchUseSemanticSearch string = 'false'
+
+@description('Semantic search config')
+param azureSearchSemanticSearchConfig string = 'default'
+
+@description('Is the index prechunked')
+param azureSearchIndexIsPrechunked string = 'false'
+
+@description('Top K results')
+param azureSearchTopK string = '5'
+
+@description('Enable in domain')
+param azureSearchEnableInDomain string = 'false'
+
+@description('Content columns')
+param azureSearchContentColumns string = 'content'
+
+@description('Filename column')
+param azureSearchFilenameColumn string = 'filename'
+
+@description('Title column')
+param azureSearchTitleColumn string = 'title'
+
+@description('Url column')
+param azureSearchUrlColumn string = 'url'
+
+@description('Name of Azure OpenAI Resource')
+param azureOpenAIResource string = '${resourcePrefix}-openai'
+
+@description('Name of Azure OpenAI Resource SKU')
+param azureOpenAISkuName string = 'S0'
+
+@description('Azure OpenAI Model Deployment Name')
+param azureOpenAIModel string = 'gpt-35-turbo'
+
+@description('Azure OpenAI Model Name')
+param azureOpenAIModelName string = 'gpt-35-turbo'
+
+param azureOpenAIModelVersion string = '0613'
+
+@description('Orchestration strategy: openai_function or langchain str. If you use a old version of turbo (0301), plese select langchain')
+@allowed([
+  'openai_function'
+  'langchain'
+])
+param orchestrationStrategy string = 'langchain'
+
+@description('Azure OpenAI Temperature')
+param azureOpenAITemperature string = '0'
+
+@description('Azure OpenAI Top P')
+param azureOpenAITopP string = '1'
+
+@description('Azure OpenAI Max Tokens')
+param azureOpenAIMaxTokens string = '1000'
+
+@description('Azure OpenAI Stop Sequence')
+param azureOpenAIStopSequence string = '\n'
+
+@description('Azure OpenAI System Message')
+param azureOpenAISystemMessage string = 'You are an AI assistant that helps people find information.'
+
+@description('Azure OpenAI Api Version')
+param azureOpenAIApiVersion string = '2023-07-01-preview'
+
+@description('Whether or not to stream responses from Azure OpenAI')
+param azureOpenAIStream string = 'true'
+
+@description('Azure OpenAI Embedding Model Deployment Name')
+param azureOpenAIEmbeddingModel string = 'text-embedding-ada-002'
+
+@description('Azure OpenAI Embedding Model Name')
+param azureOpenAIEmbeddingModelName string = 'text-embedding-ada-002'
+
+@description('Azure Cognitive Search Resource')
+param azureCognitiveSearch string = '${resourcePrefix}-search'
```

Comment:
Use "Name" in the param name if it is the name, see the azd temaplates for reference.